go-controller/pkg/node/controllers/egressip/egressip.go

```diff
@@ -1510,3 +1531,7 @@ func getNodeIPFwMarkIPRule(ipFamily int) netlink.Rule {
 	r.Family = ipFamily
 	return *r
 }
+
+func isVRFSlaveDevice(link netlink.Link) bool {
+	return link != nil && link.Attrs().Slave != nil && link.Attrs().Slave.SlaveType() == "vrf"
```

Comment:
Since `netlink.Link` is an `interface`, checking for `nil` here is a little nuanced as it will only check if the interface variable is `nil`, but not if the interface _value_ is `nil`. And if the interface _value_ is `nil`, then in the above code after this is called will try to call `link.Attrs()` anyway, so if there is a need to guard against this it might make sense to do so higher up?